src/Utilities/StaticCache.hpp

```diff
@@ -140,8 +140,15 @@ class StaticCache {
                                    Is + IndexOffset>>...}};
     // The array `cache` holds pointers to member functions, so we dereference
     // the pointer and invoke it on `this`.
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ > 10
```

Comment:
Add `&& __GNUC__ < 12` so we'll check if they fix it in each version and can remove the pragmas someday if we don't support any affected versions.